python/revng/internal/daemon/event_manager.py

```diff
@@ -57,15 +58,16 @@ def run(self):
                 self.last_event is not None and self.last_event > self.last_save
             ):
                 self.save()
-                self.last_save = time.time()
             time.sleep(10)
 
     def save(self) -> bool:
-        result = self.manager.save()
-        if result:
-            for hook in self.save_hooks:
-                hook(self.manager, cast(str, self.credentials))
-        return result
+        with self.save_lock:
+            self.last_save = time.time()
+            result = self.manager.save()
```

Comment:
This line should not be covered by `save_lock`.

Reply:
Moved out of the lock